build/script/SetVSEnvironment.cmd

```diff
@@ -13,7 +13,7 @@ if defined VSINSTALLDIR (
 
 if not exist "%TEMP%\vswhere.exe" (
   echo Downloading VSWhere so that we can find Visual Studio...
-  powershell -Command "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; Invoke-WebRequest https://github.com/microsoft/vswhere/releases/download/2.6.7/vswhere.exe -OutFile $env:TEMP\vswhere.exe" || (
+  powershell -NonInteractive -NoLogo -NoProfile -Command "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; $pp = $ProgressPreference; $ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest https://github.com/microsoft/vswhere/releases/download/2.8.4/vswhere.exe -OutFile $env:TEMP\vswhere.exe; $ProgressPreference = $pp" || (
```

Comment:
what is the purpose of $pp?

Reply:
Saves the current `$ProgressPreference` to a variable. Then, restores that value after the `Invoke-WebRequest` executes. You can use any variable name for this, so I just picked something short. See: https://thinkpowershell.com/hide-powershell-progress-bars/